applications/StructuralMechanicsApplication/tests/cpp_tests/test_metric_process.cpp

```diff
@@ -0,0 +1,351 @@
+            Properties::Pointer p_elem_prop = this_model_part.pGetProperties(0);
+            if (!KratosComponents<ConstitutiveLaw>::Has("LinearElasticPlaneStrain2DLaw"))
+                return void();
+            ConstitutiveLaw const& r_clone_cl = KratosComponents<ConstitutiveLaw>::Get("LinearElasticPlaneStrain2DLaw");
+            auto p_this_law = r_clone_cl.Clone();
+            p_elem_prop->SetValue(CONSTITUTIVE_LAW, p_this_law);
+            p_elem_prop->SetValue(YOUNG_MODULUS, 1.0);
+            p_elem_prop->SetValue(POISSON_RATIO, 0.0);
+
+            Create2DGeometry(this_model_part, "SmallDisplacementElement2D3N");
+            for (auto& ielem : this_model_part.Elements()) {
+                ielem.Initialize();
+                ielem.InitializeSolutionStep(process_info);
+            }
+            
+            const bool meshing_application_compiled = KratosComponents<Variable<Vector>>::Has("MMG_METRIC");
+            KRATOS_WARNING_IF("CalculateMetric", !meshing_application_compiled) << "Please compile the MeshingApplication in order to compute the metric" << std::endl;
+
+            // Actually computing test
+            if (meshing_application_compiled) {
+                const Variable<Vector>& metric_variable = KratosComponents<Variable<Vector>>::Get("MMG_METRIC");
+
+                // Set DISPLACEMENT_X and other variables
+                for (std::size_t i_node = 0; i_node < this_model_part.Nodes().size(); ++i_node) {
+                    auto it_node = this_model_part.Nodes().begin() + i_node;
+                    it_node->FastGetSolutionStepValue(DISPLACEMENT_X) = (it_node->X() == 1.0) ? 0.5 : 0.0;
+                    it_node->Coordinates()[0] += (it_node->X() == 1.0) ? 0.5 : 0.0;
+                    it_node->FastGetSolutionStepValue(NODAL_H) = 1.0;
+                    it_node->SetValue(metric_variable, ZeroVector(3));
+                }
+
+                // Compute metric
+                SPRMetricProcess<2> spr_process = SPRMetricProcess<2>(this_model_part);
+                spr_process.Execute();
+
+//                 // DEBUG
+//                 GiDIODebugMetricSPR(this_model_part);
+
+                const double tolerance = 1.0e-4;
+                Vector ref_metric(3);
+                ref_metric[0] = 331.893;
+                ref_metric[1] = 0;
+                ref_metric[2] = 331.893;
+
+                KRATOS_CHECK_LESS_EQUAL(norm_2(this_model_part.pGetNode(2)->GetValue(metric_variable) - ref_metric)/norm_2(ref_metric), tolerance);
+                KRATOS_CHECK_LESS_EQUAL(norm_2(this_model_part.pGetNode(3)->GetValue(metric_variable) - ref_metric)/norm_2(ref_metric), tolerance);
+            }
+        }
+        
+        /** 
+        * Checks the correct work of the nodal SPR compute
+        * Test tetrahedra
+        */
+
+        KRATOS_TEST_CASE_IN_SUITE(TestSPRMetricProcess2, KratosStructuralMechanicsFastSuite)
```

Comment:
Consider renaming tests to be more descriptive (2D / 3D). Also most of the test codes are duplicated. Could be cleanedd up.